ddl/index.go

```diff
@@ -881,13 +883,14 @@ func (w *addIndexWorker) run(d *ddlCtx) {
 		}
 
 		logutil.Logger(ddlLogCtx).Debug("[ddl] add index worker got task", zap.Int("workerID", w.id), zap.String("task", task.String()))
-		// gofail: var mockAddIndexErr bool
-		//if w.id == 0 && mockAddIndexErr && !gofailMockAddindexErrOnceGuard {
-		//	gofailMockAddindexErrOnceGuard = true
-		//	result := &addIndexResult{addedCount: 0, nextHandle: 0, err: errors.Errorf("mock add index error")}
-		//	w.resultCh <- result
-		//	continue
-		//}
+		failpoint.Inject("mockAddIndexErr", func(val failpoint.Value) {
+			if w.id == 0 && val.(bool) && !gofailMockAddindexErrOnceGuard {
```

Comment:
this case used `gofailMockAddindexErrOnceGuard` to pass the concurrent tests, Could we use WithHook to replace it?